### PR TITLE
Add support for diffing APKs

### PR DESCRIPTION
**Pull request by peterdjlee**
Add support for diffing APKs by using the newly exposed `compareProgramInfo` method in package:vm_snapshot_analysis. Need https://dart-review.googlesource.com/c/sdk/+/157080 to land first to update package dependency.

### CODE REVIEW

File: packages/devtools_app/lib/src/code_size/code_size_controller.dart [hunk 1]
```diff
@@ -129,13 +130,56 @@ class CodeSizeController {
...
+    final bool isLeafNode = json['children'] == null;
+    final node = program.makeNode(
+      name: json['n'],
+      parent: parent,
+      type: NodeType.other,
+    );
+
+    if (!isLeafNode) {
+      final List<dynamic> rawChildren = json['children'] as List<dynamic>;
+      for (Map<String, dynamic> child in rawChildren) {
```
Comment (kenzieschmoll) [inline packages/devtools_app/lib/src/code_size/code_size_controller.dart:174]: nit: rename child to childJson

Comment (peterdjlee) [reply packages/devtools_app/lib/src/code_size/code_size_controller.dart]: Done

File: packages/devtools_app/pubspec.yaml [hunk 2]
```diff
@@ -11,17 +11,16 @@ version: 0.8.1-dev.1
 homepage: https://github.com/flutter/devtools
 
 environment:
-  sdk: '>=2.6.0 <3.0.0'
+  sdk: ">=2.6.0 <3.0.0"
```
Comment (kenzieschmoll) [inline packages/devtools_app/pubspec.yaml:14]: change these back to single quotes

File: packages/devtools_app/pubspec.yaml [hunk 3]
```diff
@@ -11,17 +11,16 @@ version: 0.8.1-dev.1
...
   # The flutter desktop support interacts with build scripts on the Flutter
   # side that are not yet stable, so it requires a very recent version of
   # Flutter. This version will increase regularly as the build scripts change.
-  flutter: '>=1.10.0'
+  flutter: ">=1.10.0"
```
Comment (kenzieschmoll) [inline packages/devtools_app/pubspec.yaml:18]: ditto - can you undo any formatting changes that are unrelated to bumping the vm_snapshot_analysis version number?

File: packages/devtools_app/lib/src/code_size/code_size_controller.dart [hunk 4]
```diff
@@ -129,13 +130,56 @@ class CodeSizeController {
...
     diffMap['n'] = 'Root';
     final newRoot = generateDiffTree(diffMap);
 
     changeDiffRoot(newRoot);
   }
 
+  ProgramInfoNode _apkJsonToProgramInfo(
+    ProgramInfo program,
+    ProgramInfoNode parent,
+    Map<String, dynamic> json,
```
Comment (kenzieschmoll) [inline packages/devtools_app/lib/src/code_size/code_size_controller.dart:163]: make these required named parameters so the code reads easier

Comment (peterdjlee) [reply packages/devtools_app/lib/src/code_size/code_size_controller.dart]: Done

File: packages/devtools_app/pubspec.yaml (hunk 3 above)
Comment (peterdjlee) [reply packages/devtools_app/pubspec.yaml]: Done

File: packages/devtools_app/lib/src/code_size/code_size_controller.dart (hunk 4 above)
Comment (peterdjlee) [reply packages/devtools_app/lib/src/code_size/code_size_controller.dart]: Done
